platform/xbox/functions/_PDCLIB/_PDCLIB_fillbuffer.c

```diff
@@ -42,8 +44,7 @@ int _PDCLIB_fillbuffer( struct _PDCLIB_file_t * stream )
     }
     else
     {
-        // FIXME: Translate returned errors to proper errno
-        //_PDCLIB_errno = _PDCLIB_ERROR;
+        _PDCLIB_w32errno();
```

Comment:
@sam-itt Is this logic enough?

https://github.com/XboxDev/nxdk-pdclib/pull/22/files#diff-51228aebb715e9137534cdecdb53aa0dR53 does a bit more and you said `feof` was broken (or is this fixed independently in this PR already?).

Are there any other changes in your PR that are different than what is done here? Could you please leave review comments asap?

Reply:
feof is currently broken because XReadFile doesn't sets the flag and doesn't give enough info to do it at the _PDCLIB_fillbuffer level. 

You *need* to detect EOF and set stream->status  accordingly here. Otherwise it won't be set and thus feof won't work.

Reply:
EOF is already handled: https://github.com/thrimbor/nxdk-pdclib/blob/fcd7865af33af10c3b5ab31139d79d5b2fee8282/platform/xbox/functions/_PDCLIB/_PDCLIB_fillbuffer.c#L28-L33

The additional EOF-handling code in #22 is never actually used, because `ReadFile` doesn't report EOF this way - it only does that when using overlapped IO, which we don't do here (nxdk doesn't even support that yet). When using synchronous IO, `ReadFile` reports EOF by returning `TRUE` while setting `lpNumberOfBytesRead` to zero.

If you have any issues regarding EOF handling, please provide a minimal test case.

Reply:
feof() doesn't work with the current X*File* based code. If it works with your ReadFile-based implementation that's allright by me.